### DISCUSSION TITLE
Retrieving data from automation

### DISCUSSION THREAD

**Discussion by joeytzy** [Budibase/budibase#11534] [Help] [answered]
Hi, so I tried making an  automation to handle custom validation and the last step of the validation is js scripting. I was wondering if it is possible to pass a return value from js scripting in the automation and retrieve it outside of the automation such as retrieving the value in screen. Basically, a screen with on screen load have 2 action. First is the trigger automation and second is the update state where the value is from the automation when running the trigger automation.

**Comment by melohagan** [comment, marked as answer]
Hey @joeytzy 

This is possible with [Sync automations](https://docs.budibase.com/docs/synchronous-automations)

**Reply by joeytzy** [reply]
i see, so by default the user is not able to access the automation and is only able to access it through synchronous automations

**Reply by melohagan** [reply]
That's right, most automations typically run as background jobs.
